### PR TITLE
Update citadel and off-gcp overlays for multicloud support

### PR DESCRIPTION
**Pull request by backward-compatible**

### CODE REVIEW

File: asm/istio/options/citadel-ca.yaml [hunk 1]
```diff
@@ -28,8 +28,8 @@ spec:
     trustDomain: "cluster.local"
     defaultConfig:
       proxyMetadata:
-        GKE_CLUSTER_URL: "" # {"$ref":"#/definitions/io.k8s.cli.substitutions.gke-cluster-url"}
-        GCP_METADATA: "" # {"$ref":"#/definitions/io.k8s.cli.substitutions.gke-metadata"}
+        GKE_CLUSTER_URL: ""
+        GCP_METADATA: ""
```
Comment (zerobfd) [inline asm/istio/options/citadel-ca.yaml:32]: I bet the right thing to do is actually remove these two completely from citadel-ca and move them to off-gcp without the refs.

**Review by zerobfd — APPROVED** [review]

File: asm/istio/options/citadel-ca.yaml (hunk 1 above)
Comment (backward-compatible) [reply asm/istio/options/citadel-ca.yaml:32]: ok. would you say the same about **GCE_METADATA_HOST** field too (expand citadel-ca.yaml)?

Comment (zerobfd) [reply asm/istio/options/citadel-ca.yaml:32]: Yep, good catch

Comment (backward-compatible) [reply asm/istio/options/citadel-ca.yaml]: done

**Pull request merged**